### PR TITLE
[Merged by Bors] - feat(NumberTheory.NumberField.CanonicalEmbedding): add exists_ne_zero_mem_ringOfIntegers_lt

### PR DESCRIPTION
**Pull request by xroblot**
This PR proves the following result: 
```lean
theorem exists_ne_zero_mem_ringOfIntegers_lt (h : minkowski_bound K < volume (convex_body_lt K f)) :
     ∃ (a : 𝓞 K), a ≠ 0 ∧ ∀ w : InfinitePlace K, w a < f w 
```
where `f : InfinitePlace K → ℝ≥0` and `convex_body_lt K f` is the set of points `x` such that `‖x w‖ < f w` for all infinite places `w`. This is a key result in the proof of Dirichlet's unit theorem #5960

---

- [x] depends on: #5518 
- [x] depends on: #5671 
- [x] depends on: #5728 
- [x] depends on: #6832 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5518
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean [hunk 1]
```diff
@@ -145,3 +158,275 @@ theorem mem_span_latticeBasis [NumberField K] (x : (K →+* ℂ) → ℂ) :
...
+
+open Classical Fintype MeasureTheory MeasureTheory.Measure BigOperators
+
+-- See: https://github.com/leanprover/lean4/issues/2220
+local macro_rules | `($x ^ $y) => `(HPow.hPow $x $y)
+
+variable [NumberField K]
+
+/-- The fudge factor that appears in the formula for the volume of `convex_body_lt`. -/
+noncomputable abbrev constant_factor_lt : ℝ≥0∞ :=
```
Comment (jcommelin) [inline Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean:328]: Similar remark applies.

File: Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean [hunk 2]
```diff
@@ -145,3 +158,275 @@ theorem mem_span_latticeBasis [NumberField K] (x : (K →+* ℂ) → ℂ) :
...
+
+section minkowski
+
+open MeasureTheory MeasureTheory.Measure Classical NNReal ENNReal FiniteDimensional Zspan
+
+variable [NumberField K]
+
+/-- The bound that appears in Minkowski theorem, see
+`MeasureTheory.exists_ne_zero_mem_lattice_of_measure_mul_two_pow_lt_measure`. -/
+noncomputable def minkowski_bound : ℝ≥0∞ :=
```
[the remark this answers is not in the document]
Comment (xroblot) [reply Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean]: Thanks for the review. I tried to use better names.

**Comment by jcommelin** [comment]
I think this is looking pretty good now. If you fix the two naming issues that I pointed out above, please kick this on the bors queue.

bors d+

**Pull request closed without merging**